### PR TITLE
Contract adaptor

### PR DESCRIPTION
**Pull request by tkmct**
Add Universal decision contract adaptor

### CODE REVIEW

**Comment by tkmct** [comment]
Build failing because of db crate error?

File: ovm/src/types/core.rs [hunk 1]
```diff
@@ -86,6 +86,12 @@ impl Encodable for Property {
     }
 }
 
+impl From<Property> for Token {
+    fn from(property: Property) -> Token {
+        Token::Tuple(property.to_tuple())
+    }
+}
```
Comment (syuhei176) [inline ovm/src/types/core.rs]: This is interesting!

**Comment by tkmct** [comment]
@syuhei176 
I've got two errors when I tried to compile OVM crate. Do you have any idea?

```
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> ovm/src/deciders/preimage_exists_decider.rs:154:22
    |
154 |           let result = decider
    |  ______________________^
155 | |             .get_db()
156 | |             .bucket(&BaseDbKey::from(&b"preimage_exists_decider"[..]))
157 | |             .get(&BaseDbKey::from(decision_key.as_bytes()))
158 | |             .map_err::<Error, _>(Into::into)?;
    | |_____________________________________________^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: required by `std::option::Option`
```

```
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
   --> ovm/src/deciders/signed_by_decider.rs:148:16
    |
148 |         if let Some(decision_value_bytes) = result {
    |                ^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `std::marker::Sized` is not implemented for `[u8]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: required by `std::prelude::v1::Some`
```

**Comment by syuhei176** [comment]
> @syuhei176
> I've got two errors when I tried to compile OVM crate. Do you have any idea?
> 
> ```
> error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
>    --> ovm/src/deciders/preimage_exists_decider.rs:154:22
>     |
> 154 |           let result = decider
>     |  ______________________^
> 155 | |             .get_db()
> 156 | |             .bucket(&BaseDbKey::from(&b"preimage_exists_decider"[..]))
> 157 | |             .get(&BaseDbKey::from(decision_key.as_bytes()))
> 158 | |             .map_err::<Error, _>(Into::into)?;
>     | |_____________________________________________^ doesn't have a size known at compile-time
>     |
>     = help: the trait `std::marker::Sized` is not implemented for `[u8]`
>     = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
>     = note: required by `std::option::Option`
> ```
> 
> ```
> error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
>    --> ovm/src/deciders/signed_by_decider.rs:148:16
>     |
> 148 |         if let Some(decision_value_bytes) = result {
>     |                ^^^^^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
>     |
>     = help: the trait `std::marker::Sized` is not implemented for `[u8]`
>     = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
>     = note: required by `std::prelude::v1::Some`
> ```

compiler version?

This is mine

>  (contract-adaptor)$ rustc --version
rustc 1.35.0 (3c235d560 2019-05-20)

**Comment by tkmct** [comment]
Here is mine.
`rustc 1.35.0 (3c235d560 2019-05-20)`

File: contract-wrapper/UniversalDecisionContract.json [hunk 2]
```diff
@@ -0,0 +1,436 @@
+[
```
Comment (syuhei176) [inline contract-wrapper/UniversalDecisionContract.json]: This is from https://github.com/cryptoeconomicslab/ovm-contracts.
Thanks!

**Comment by tkmct** [comment]
Thanks!!
Okay, that makes whole code base more simplified.

**Pull request merged**